### PR TITLE
Storage folder refactoring 

### PR DESCRIPTION
**Pull request by JamesLiuZX**
- Refactored AddressBook, addressBook -> RecipeBook, recipeBook across all folders. 
- Refactored Person, person -> Recipe, recipe 
- Refactored JsonAdaptedRecipe (wrote some docs for explanation)

- [x] Merge conflicts

### CODE REVIEW

File: src/main/java/seedu/recipe/logic/commands/ListCommand.java
Comment (alson001) [inline src/main/java/seedu/recipe/logic/commands/ListCommand.java:1]: recipe book

File: src/main/java/seedu/recipe/storage/JsonSerializableRecipeBook.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import seedu.recipe.commons.exceptions.IllegalValueException;
+import seedu.recipe.model.RecipeBook;
+import seedu.recipe.model.ReadOnlyRecipeBook;
+import seedu.recipe.model.recipe.Recipe;
+
+/**
+ * An Immutable RecipeBook that is serializable to JSON format.
+ */
+@JsonRootName(value = "addressbook")
```
Comment (alson001) [inline src/main/java/seedu/recipe/storage/JsonSerializableRecipeBook.java:19]: recipebook but maybe not if json still reads addressbook

**Review by alson001 — APPROVED** [review]

File: src/main/java/seedu/recipe/storage/JsonSerializableRecipeBook.java (hunk 1 above)
Comment (ZHTang29) [reply src/main/java/seedu/recipe/storage/JsonSerializableRecipeBook.java]: Changes will be done on my end in model.UserPrefs - approved

File: src/main/java/seedu/recipe/model/recipe/RecipeIngredient.java [hunk 2]
```diff
@@ -48,7 +48,8 @@ public int hashCode() {
     @Override
     public String toString() {
         StringBuilder sb = new StringBuilder();
-        sb.append(amount).append(" ").append(" ").append(unit).append(name).append("/").append(substitutions.toString());
+        sb.append(amount).append(" ").append(unit).append(" ")
+        .append(name).append("/").append(substitutions.toString());
         return sb.toString();
```
Comment (SeeuSim) [inline src/main/java/seedu/recipe/model/recipe/RecipeIngredient.java:53]:
```java
String fmt = String.format(
    "%s %s %s/%s,
    amount, unit, name, substitutions
);
return out;
```

File: src/main/java/seedu/recipe/storage/JsonAdaptedRecipe.java [hunk 3]
```diff
@@ -3,14 +3,15 @@
...
+import java.util.Optional;
 import java.util.Set;
-import java.util.stream.Collectors;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import seedu.recipe.commons.exceptions.IllegalValueException;
 import seedu.recipe.model.recipe.*;
+import seedu.recipe.model.recipe.unit.*;
```
Comment (SeeuSim) [inline src/main/java/seedu/recipe/storage/JsonAdaptedRecipe.java:14]: wildcard imports

File: src/main/java/seedu/recipe/storage/JsonAdaptedRecipe.java [hunk 4]
```diff
@@ -19,61 +20,47 @@
 class JsonAdaptedRecipe {
 
     public static final String MISSING_FIELD_MESSAGE_FORMAT = "Recipe's %s field is missing!";
+    private String name;
+    private String portion;
+    private String duration;
+    private String tags;
```
Comment (SeeuSim) [inline src/main/java/seedu/recipe/storage/JsonAdaptedRecipe.java:26]: Might be appropriate to store these fields as per the old method, using a list of `JsonAdaptedXXX`

File: src/main/java/seedu/recipe/storage/JsonAdaptedRecipe.java [hunk 5]
```diff
@@ -82,41 +69,69 @@ public Recipe toModelType() throws IllegalValueException {
...
+        if (!output.isEmpty()) {
+            final Optional<Set<Tag>> optionalModelTags = Optional.of(output);
         }
-        final Step modelStep = new Step(step);
-
-        final Set<Tag> modelTags = new HashSet<>(recipeTags);
-        return new Recipe(modelName, modelIngredient, modelEmail, modelAddress, modelTags, modelStep);
+        Recipe res = new Recipe(modelName);
+        //Setter methods/overloaded constructor
+        return res;
```
Comment (SeeuSim) [inline src/main/java/seedu/recipe/storage/JsonAdaptedRecipe.java:134]: SLAP and design can be *a lot* better

- Each model can have a `toDbSchema` method and inherit from a `JSONAdaptable` interface
- In the constructor for this, simply pass in the field objects natively and call the `toDbSchema` as opposed to overloading the `toString`
  - Can also have these fields be an `Optional` to set the JSON if present
    - RecipePortion
    - RecipeDuration
  
- Each model can also have a `fromDbSchema`

**Pull request merged**